DependencyInjection/Compiler/ResourcePass.php

```diff
@@ -42,19 +42,15 @@ public function process(ContainerBuilder $container)
                 continue;
             }
 
+            $isAbstract = $this->isAbstract($resourceDefinition->getArgument(0));
             if (!$resourceDefinition->hasMethodCall('initItemOperations')) {
-                $resourceDefinition->addMethodCall('initItemOperations', [[
-                    $this->createOperation($container, $serviceId, 'GET', false),
-                    $this->createOperation($container, $serviceId, 'PUT', false),
-                    $this->createOperation($container, $serviceId, 'DELETE', false),
-                ]]);
+                $methods = $isAbstract ? ['GET', 'DELETE'] : ['GET', 'PUT', 'DELETE'];
```

Comment:
Methods can be used as keys to avoid the `array_map` call and reduce the complexity O(1) instead of O(n).